cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -473,6 +473,35 @@ func (o *objectGraph) Discovery(namespace string) error {
 	// Completes the graph by setting for each node the list of tenants the node belongs to.
 	o.setTenants()
 
+	// Filter and remove nodes in the graph that do not belong to cluster
+	if cluster != "" {
+		return o.filterCluster(cluster)
+	}
+
+	return nil
+}
+
+// filterCluster removes all objects but provided cluster and its dependents and soft-dependents
+func (o *objectGraph) filterCluster(cluster string) error {
+	for _, node := range o.getNodes() {
+		clusterTenant := false
```

Comment:
What do you think about storing here the name of the cluster?
That way the error can say something like `"resource blabla is a dependent of clusters bla1 and bla2. Only one cluster owner is allowed"`